internal/venice-common/src/main/java/com/linkedin/venice/utils/PartitionUtils.java

```diff
@@ -23,41 +24,40 @@ public class PartitionUtils {
   private static final Logger LOGGER = LogManager.getLogger(PartitionUtils.class);
 
   /**
-   * Calculate partition count for new version. If the version is the first one of the given store,
-   * calculate the number by given store size and partition size. Otherwise, use the number from the current active
-   * version.
+   * Calculate partition count for new version. If store level partition count is not configured (0),
+   * calculate the number by storage quota and partition size. Otherwise, use the store level partition count.
    */
   public static int calculatePartitionCount(
```

Comment:
BTW, we always say hybrid store partition count is immutable, which means Controller shouldn't recalculate the partition count for every new push after the first push.
Can you help me find that piece of logic? Somehow I can't find it.

Reply:
Currently there's no logic to update store-level partition count after the first push.

Hybrid store partition count is immutable because Nuage updates store-level partition count when a store is converted to hybrid. Since store-level partition count is set, we won't recalculate partition count for new pushes.

Reply:
What if some hybrid store is converted in prod directly? That means the partition count won't be setup at the promotion time, so the behavior will be similar as the regular batch-only store, right?

Reply:
Yeah that's a bug in the current system. I tested by converting a new store in prod from batch-only to hybrid. The store-level partition count keeps as 0. The first empty push created VT and RT with 20 partitions. Later empty push with 300GB input size created VT with 30 partitions but RT remains 20 partitions.

Reply:
The bug is fixed by another PR https://github.com/linkedin/venice/pull/146 Now controller always enforces a default partition count for hybrid stores.